CMakeModules/FindNCCL.cmake

```diff
@@ -0,0 +1,24 @@
+# Find the NCCL libraries
+#
+# The following variables are optionally searched for defaults
+#  NCCL_ROOT_DIR:    Base directory where all NCCL components are found
+#
+# The following are set after configuration is done:
+#  NCCL_FOUND
+#  NCCL_INCLUDE_DIR
+#  NCCL_LIBRARY
+
+find_path(NCCL_INCLUDE_DIR NAMES nccl.h
+    PATHS ${NCCL_ROOT_DIR}/include
```

Comment:
User can define the extra environmental `NCCL_ROOT_DIR` to show cmake where to search for in addition. Also, system path and `PATH` variable are searched anyways. So supposing we are on a unix/linux system and user has installed NCCL (or whatever in a certain \<dir\> which has bin/include/lib ), it suffices to have \<dir\>/bin included in `PATH`. And usually in order for a shell to find executables, user (or the system for him) has already properly defined the `PATH` to include at least '\<dir\>/cuda', '/usr/local/bin' and '/usr/bin'.

See [find_path documentation](https://cmake.org/cmake/help/v3.4/command/find_path.html):  
> 4. Search the standard system environment variables. This can be skipped if NO_SYSTEM_ENVIRONMENT_PATH is an argument.
>    * Directories in INCLUDE, <prefix>/include/<arch> if CMAKE_LIBRARY_ARCHITECTURE is set, and <prefix>/include for each <prefix>/[s]bin in PATH, and <entry>/include for other entries in PATH, and the directories in PATH itself."

So, I think this is fine. By the way I found this [here](https://github.com/NVIDIA/caffe/blob/caffe-0.15/cmake/Modules/FindNCCL.cmake).